src/test/components/WindowTitle.test.js

```diff
@@ -124,4 +130,44 @@ describe('WindowTitle', () => {
 
     expect(document.title).toBe('Compare Profiles – Firefox Profiler');
   });
+
+  it("shows a title when a profile isn't loaded yet", () => {
+    const store = blankStore();
+    store.dispatch(setDataSource('from-url'));
+    render(
+      <Provider store={store}>
+        <WindowTitle />
+      </Provider>
+    );
+
+    expect(document.title).toBe('Firefox Profiler');
+  });
+
+  it('shows the file name when viewing a file from a zip file', async () => {
+    const { store } = await storeWithZipFile([
+      'foo/bar/profile1.json',
+      'foo/profile2.json',
+      'baz/profile3.json',
+    ]);
+
+    store.dispatch(setDataSource('from-url'));
+
+    render(
+      <Provider store={store}>
+        <WindowTitle />
+      </Provider>
+    );
+
+    expect(document.title).toBe('Zip File Contents – Firefox Profiler');
+
+    await store.dispatch(
+      ZippedProfilesActions.viewProfileFromPathInZipFile(
+        'foo/bar/profile1.json'
+      )
+    );
+
+    expect(document.title).toBe(
+      'bar/profile1.json – Firefox – 1/1/1970, 12:00:00 AM UTC – Firefox Profiler'
+    );
+  });
```

Comment:
I checked that these 2 new testcase are failing with an exception on the main branch.